src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, (e) => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
```

Comment:
Bug: Your waitFor function will only resolve the promise the first time the event is fired. If you want to handle every event, you need to create a new promise for each event. Alternatively, you could remove the event listener once the promise has resolved to avoid memory leaks.